src/Examples/UnitModels/Operations/skeleton_unit_testing.ipynb

```diff
@@ -0,0 +1,420 @@
+      "        inequality constraints with only upper bounds:        0\n",
+      "\n",
+      "iter    objective    inf_pr   inf_du lg(mu)  ||d||  lg(rg) alpha_du alpha_pr  ls\n",
+      "   0  0.0000000e+00 0.00e+00 0.00e+00  -1.0 0.00e+00    -  0.00e+00 0.00e+00   0\n",
+      "\n",
+      "Number of Iterations....: 0\n",
+      "\n",
+      "                                   (scaled)                 (unscaled)\n",
+      "Objective...............:   0.0000000000000000e+00    0.0000000000000000e+00\n",
+      "Dual infeasibility......:   0.0000000000000000e+00    0.0000000000000000e+00\n",
+      "Constraint violation....:   0.0000000000000000e+00    0.0000000000000000e+00\n",
+      "Complementarity.........:   0.0000000000000000e+00    0.0000000000000000e+00\n",
+      "Overall NLP error.......:   0.0000000000000000e+00    0.0000000000000000e+00\n",
+      "\n",
+      "\n",
+      "Number of objective function evaluations             = 1\n",
+      "Number of objective gradient evaluations             = 1\n",
+      "Number of equality constraint evaluations            = 1\n",
+      "Number of inequality constraint evaluations          = 0\n",
+      "Number of equality constraint Jacobian evaluations   = 1\n",
+      "Number of inequality constraint Jacobian evaluations = 0\n",
+      "Number of Lagrangian Hessian evaluations             = 0\n",
+      "Total CPU secs in IPOPT (w/o function evaluations)   =      0.000\n",
+      "Total CPU secs in NLP function evaluations           =      0.000\n",
+      "\n",
+      "EXIT: Optimal Solution Found.\n"
+     ]
+    }
+   ],
+   "source": [
+    "m.fs.skeleton.inlet.flow_mol_comp[0, \"c1\"].fix(2*pyunits.mol/pyunits.s)\n",
+    "m.fs.skeleton.inlet.flow_mol_comp[0, \"c2\"].fix(2*pyunits.mol/pyunits.s)\n",
+    "m.fs.skeleton.inlet.temperature.fix(325*pyunits.K)\n",
+    "m.fs.skeleton.inlet.pressure.fix(200*pyunits.kPa)\n",
+    "\n",
+    "assert degrees_of_freedom(m) == 0\n",
+    "\n",
+    "m.fs.skeleton.initialize()\n",
+    "\n",
+    "solver = SolverFactory('ipopt')\n",
+    "results = solver.solve(m, tee = True)\n",
+    "\n",
+    "assert check_optimal_termination(results)\n",
+    "assert_units_consistent(m)\n",
+    "\n",
+    "assert value(m.fs.skeleton.outlet.flow_mol_comp[0, \"c1\"]) == pytest.approx(2, abs=1e-3)\n",
+    "assert value(m.fs.skeleton.outlet.flow_mol_comp[0, \"c2\"]) == pytest.approx(2, abs=1e-3)\n",
+    "assert value(m.fs.skeleton.outlet.temperature[0]) == pytest.approx(335, abs=1e-3)\n",
+    "assert value(m.fs.skeleton.outlet.pressure[0]) == pytest.approx(190, abs=1e-3)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The Skeleton model allows for a custom, user-defined initialization scheme as well, as demonstrated below. Users can similarly introduce custom variable and constraint scaling using IDAES scaling tools (not demonstrated in this example):"
```

Comment:
This needs far more discussion and explanation. What are you doing here, what is different to the first example and what does it demonstrate to the user?